packages/web/src/fauceting/RequestFunds.tsx

```diff
@@ -0,0 +1,185 @@
+import * as React from 'react'
+import ReCAPTCHA from 'react-google-recaptcha'
+import { StyleSheet, View } from 'react-native'
+import { ButtonWithFeedback, ContextualInfo, HashingStatus } from 'src/fauceting/MicroComponents'
+import {
+  formatNumber,
+  getCaptchaKey,
+  RequestState,
+  requestStatusToState,
+  validateBeneficary,
+} from 'src/fauceting/utils'
+import { TextInput } from 'src/forms/FormComponents'
+import { I18nProps, NameSpaces, withNamespaces } from 'src/i18n'
+import { postForm } from 'src/shared/Form'
+import { colors, standardStyles } from 'src/styles'
+import { RequestRecord, RequestType, subscribeRequest } from '../../server/FirebaseClient'
+
+function send(beneficiary: string, kind: RequestType, captchaToken: string) {
+  const route = kind === RequestType.Invite ? '/invite' : '/faucet'
+  return postForm(route, { captchaToken, beneficiary })
+}
+
+interface State {
+  beneficiary: string
+  captchaOK: boolean
+  requestState: RequestState
+  dollarTxHash?: string | null
+  goldTxHash?: string | null
+  escrowTxHash?: string | null
+}
+
+interface Props {
+  kind: RequestType
+}
+
+class RequestFunds extends React.PureComponent<Props & I18nProps, State> {
+  state: State = {
+    beneficiary: '',
+    requestState: RequestState.Initial,
+    captchaOK: false,
+  }
+
+  recaptchaRef = React.createRef<ReCAPTCHA>()
+
+  setBeneficiary = ({ nativeEvent }) => {
+    const { value } = nativeEvent.target
+    const beneficiary = this.props.kind === RequestType.Invite ? formatNumber(value) : value
+    this.setState({
+      beneficiary,
+      requestState:
+        this.state.requestState !== RequestState.Working
+          ? RequestState.Initial
+          : this.state.requestState,
+    })
+  }
+
+  onCaptcha = (value: string | null) => {
+    this.setState({ captchaOK: !!value })
+  }
+  resetCaptcha = () => {
+    this.recaptchaRef.current.reset()
+  }
+  getCaptchaToken = () => {
+    return this.recaptchaRef.current.getValue()
+  }
+
+  checkCheckSum = () => {
+    // TODO check checksum if bad this.setState({requestState: RequestState.BadChecksum}) but dont abort
```

Comment:
this check can be done with `checkAddressChecksum` from `web3-utils`

Reply:
the more i think about this the more im wondering how important it is to do a specific checksum check.

Reply:
We could ignore it. `isAddress` from web3-utils could also work well (it only tests the checksum if it thinks the address is checksum'd, if it's not checksum'd, it just checks the length)